arviz/plots/backends/matplotlib/posteriorplot.py

```diff
@@ -319,18 +319,23 @@ def format_axes():
             rug=False,
             show=False,
         )
-    else:
+    elif values.dtype.kind == "i":
```

Comment:
the initial condition is `if kind=kde and dtype=f` so this condition is too restrictive, I think it should be `if dtype=i or (dtype=f and kind=hist)`

Reply:
Yea right. I should have done it that way. Thanks for pointing.

Reply:
test exists because we all do this which is also why writing tests is important everywhere

Reply:
Done.
One more observation. Now, that there is an added kind of plot `bar` for boolean values. So maybe we should set 
`kind = 'bar'` somewhere and should accept `kind = 'bar'`.